lib/artirix_data_models/cached_action_adaptor.rb

```diff
@@ -23,7 +23,7 @@ def fetch(&block)
     elsif block_given?
       perform &block
     else
-      nil
+      raise "Unexpected NilClass returned from cache - case 1"
```

Comment:
this line is never called. The only place that we do a `cache_adaptor.call` is with a block, and it is in the `DataGateway`'s `call` method.